### PR TITLE
IG-1881 [Nuclio] Add left navigation menu

### PR DESCRIPTION
**Pull request by mavdryk**

### CODE REVIEW

File: pkg/dashboard/ui/src/app/components/data-wrappers/projects-data-wrapper/projects-data-wrapper.component.js [hunk 1]
```diff
@@ -17,6 +17,16 @@
         ctrl.updateProject = updateProject;
         ctrl.getProjects = getProjects;
 
+        ctrl.$onInit = onInit;
+
+        //
+        // Hook methods
+        //
+
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/data-wrappers/projects-data-wrapper/projects-data-wrapper.component.js]: JSDoc?

File: pkg/dashboard/ui/src/app/components/data-wrappers/projects-data-wrapper/projects-data-wrapper.component.js [hunk 2]
```diff
@@ -17,6 +17,16 @@
         ctrl.updateProject = updateProject;
         ctrl.getProjects = getProjects;
 
+        ctrl.$onInit = onInit;
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/data-wrappers/projects-data-wrapper/projects-data-wrapper.component.js]: Hook methods before public(sorting)

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        var ctrl = this;
+
+        ctrl.selectedNamespace = null;
+        ctrl.namespaces = [];
+
+        ctrl.isActiveState = isActiveState;
+        ctrl.resetNamespaceDropdown = resetNamespaceDropdown;
+        ctrl.onDataChange = onDataChange;
+        ctrl.isDemoMode = ConfigService.isDemoMode;
+        ctrl.isStagingMode = ConfigService.isStagingMode;
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: a-z sorting

Comment (YarovoiVadym) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: nuclio doesn't have 'staging' mode

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        ctrl.$onInit = onInit;
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: hook method before public(sorting)

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+        //
+        // Hook methods
+        //
+
+        function onInit() {
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: JSDoc

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+            NuclioProjectsDataService.getNamespaces()
+                .then(function (response) {
+                    lodash.forEach(response.namespaces.names, function (name) {
+                        ctrl.namespaces.push({
+                            type: 'namespace',
+                            id: name,
+                            name: name
+                        });
+                    });
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: please use `lodash.map`

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                })
+                .catch(function (error) {
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: I think you can remove `error` from arguments if you don't use it

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+            padding: 0 15px;
+            font-size: 16px;
+
+            .namespace-label {
+                color: @light-grey-blue;
+                margin-bottom: 10px;
+            }
+
+            .default-dropdown {
+
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: please remove blank line

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+                .default-dropdown-field {
+                    color: rgba(255, 255, 255, 0.84);
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: please use
```
.white(0.84); or .nclWhite(0.84);
color: @color;
```

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less (hunk 8 above)
Comment (YarovoiVadym) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: please remove empty line

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+                    height: 40px;
+                    background-color: @dusk;
+                    border: none;
+                    border-radius: 3px;
+
+                    input::placeholder {
+                        color: rgba(255, 255, 255, 0.84);
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: see above

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+                }
+
+                .default-dropdown-container {
+                    margin-top: -5px;
+                    background-color: @dusk;
+                    border: none;
+                    border-radius: 3px;
+
+                    .list-item {
+                        color: rgba(255, 255, 255, 0.84);
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: see above

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+                        &.active {
+                            color: #f9794f;
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.less]: I think we need to declare colors as variables (in all places)

File: pkg/dashboard/ui/src/app/shared/services/nuclio-projects-data.service.js [hunk 13]
```diff
@@ -11,7 +11,8 @@
             getExternalIPAddresses: getExternalIPAddresses,
             getProject: getProject,
             getProjects: getProjects,
-            updateProject: updateProject
+            updateProject: updateProject,
+            getNamespaces: getNamespaces
```
Comment (YarovoiVadym) [inline pkg/dashboard/ui/src/app/shared/services/nuclio-projects-data.service.js]: please sort a-z

File: pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        //
+        // Public method
+        //
+
+        /**
+         * Checks if namespaces exist
+         * @returns {boolean}
+         */
+        function isNamespacesExist() {
+            return !lodash.isEmpty(ctrl.namespaceData.namespaces)
```
Comment (maksym-kyselytsia) [inline pkg/dashboard/ui/src/app/components/main-side-menu/main-side-menu.component.js]: missed `;`

**Pull request closed without merging**